native/jsonrs/src/lib.rs

```diff
@@ -1,10 +1,11 @@
 use rustler::{Env, Error, Term};
+mod compression;
 
 rustler::init!("Elixir.Jsonrs", [encode, decode]);
 
 #[rustler::nif(name = "nif_encode", schedule = "DirtyCpu")]
-fn encode(term: Term, indent_size: Option<u32>) -> Result<String, Error> {
-  let mut buf = Vec::new();
+fn encode(term: Term, indent_size: Option<u32>, compression: Option<(compression::Algs, Option<u32>)>) -> Result<String, Error> {
+  let mut buf = compression::get_writer(compression);
```

Comment:
Just to disambiguate this against the module name:
```suggestion
fn encode(term: Term, indent_size: Option<u32>, comp_opts: Option<(compression::Algs, Option<u32>)>) -> Result<String, Error> {
  let mut buf = compression::get_writer(comp_opts);
```